### PR TITLE
PLANNER-2207 Refactor maintenance scheduling backend to repository pattern

### PR DESCRIPTION
**Pull request by cuijulian**
Jira: https://issues.redhat.com/browse/PLANNER-2207

### CODE REVIEW

File: quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java [hunk 1]
```diff
@@ -71,6 +69,44 @@ public MaintenanceJob(String jobName, MaintainableUnit maintainableUnit, int rea
         this.critical = critical;
     }
 
+    // ************************************************************************
+    // Complex methods
+    // ************************************************************************
+
+    public int calculateOverlap(MaintenanceJob other) {
```
Comment (triceo) [inline quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java:76]: Could use a test or two. (Or did I overlook them?)

File: quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java [hunk 2]
```diff
@@ -71,6 +69,44 @@ public MaintenanceJob(String jobName, MaintainableUnit maintainableUnit, int rea
...
+            return 0;
+        } else if (otherEnd < start) {
+            return 0;
+        }
+        return Math.min(end, otherEnd) - Math.max(start, otherStart);
+    }
+
+    @Override
+    public String toString() {
+        return "MaintenanceJob{" +
```
Comment (triceo) [inline quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java:95]: Any plan to show these strings anywhere in the UI? If so, they should be much less verbose.
If they're purely for debugging, this is fine.

**Review by triceo — APPROVED** [review]
LGTM, just minor comments.
Also, please file a separate JIRA for this, so that there's a record.

File: quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java (hunk 1 above)
Comment (cuijulian) [reply quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java:76]: It's covered in the constraint provider tests ([example](https://github.com/kiegroup/optaplanner-quickstarts/blob/d46c8aea02612a0549ead6de62e2a5825a510f9d/quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/solver/MaintenanceScheduleConstraintProvider.java#L84)).

File: quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java (hunk 2 above)
Comment (cuijulian) [reply quarkus-maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceJob.java:95]: No plans to show this in UI. I'm keeping them for debugging purposes for now, but will probably remove them in a final cleanup.

**Pull request merged**